cli/EAP7-1100-Add_the_ability_for_the_CLI_SSL_security_commands_to_be_able_to_obtain_a_server_certificate_from_Lets_Encrypt.adoc

```diff
@@ -0,0 +1,120 @@
+= EAP7-1100 Add the ability for the CLI SSL security commands to be able to obtain a server certificate from Let's Encrypt
+:author:            Marek Marusic
+:email:             mmarusic@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          cli,add,Let's,Encrypt,Let's Encrypt,security,commands
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WFCORE-3447 introduced high-level security commands (ssl enable-ssl-management and ssl enable-ssl-http-server)
+for enabling one-way and two-way SSL easily.
+To generate the server certificate,
+these high-level security commands made use of key-store management operations to generate a self-signed certificate.
+Since it is now possible to obtain certificates from the Let's Encrypt certificate authority using the CLI,
+the high-level security commands should be updated so that in addition to being able to generate a self-signed server certificate,
+they can also obtain a server certificate from Let's Encrypt.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1100[EAP7-1100]
+* https://issues.jboss.org/browse/WFCORE-4227[WFCORE-4227]
+
+=== Related Issues
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Affected Projects or Components
+
+* CLI
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+* Add parameter '--lets-encrypt' to the 'enable-ssl-http-server' and 'enable-ssl-management' security commands
+* The parameter '--lets-encrypt' should be used in combination with the '--interactive' parameter
+* After '--lets-encrypt parameter' is used user is asked for several information needed to obtain the certificate:
+** Key-store file name
+** Password
```

Comment:
I think two separate password options would be needed (one for the certificate key-store and one for the account key-store).

Reply:
What about alias ? Should we have 2 separate aliases as well?

Reply:
Yes, good point. Two separate aliases would be needed as well. Like with the other options, if reusing a certificate authority account, the user won't need to be asked to provide the account key-store alias.

Reply:
I just noticed, the alias is used for the certificate not for the key-store. Sorry for the confusion.

Reply:
When creating a ```certificate-authority-account```, an alias is needed as well (this is the alias that will store the user's Let's Encrypt account key), e.g., 

```
/subsystem=elytron/certificate-authority-account=myLetsEncryptAccount:add(alias=letsEncrypt,key-store=accountsKS,contact-urls=[mailto:admin@admin.org])  
```

And then a separate alias is used when obtaining the certificate to indicate the alias in the ```key-store``` that will be used to store the certificate, e.g.,

```
/subsystem=elytron/key-store=serverKS:obtain-certificate(alias=server,domain-names=[www.example.org],certificate-authority-account=myLetsEncryptAccount,agree-to-terms-of-service)  
```

Reply:
Thank you for the clarification!